lib/stages/FloatPhaseUpdate.cpp

```diff
@@ -11,13 +11,87 @@ FloatPhaseUpdate::FloatPhaseUpdate(kotekan::Config& config, const std::string& u
 void FloatPhaseUpdate::compute_phases(uint8_t* out_frame, const timespec& gps_time,
                                       const std::vector<freq_id_t>& frequencies_in_frame,
                                       uint8_t* gains_frame) {
-    // These lines are just to suppress warnings, remove once function uses them
-    (void)out_frame;
-    (void)gps_time;
-    (void)frequencies_in_frame;
-    (void)gains_frame;
-
-    // Code to generate phases goes here.
-    // Can access configuration parameters from BeamformingPhaseUpdate
-    // e.g. _inst_lat and _inst_long, _beam_coord, _num_beams, etc.
+    //Getting UTC time in ISO format from the GPS time in Unix format
+    struct tm* utc_time;
+    utc_time = gmtime(&gps_time.tv_sec);                                                                      //DOUBT: Confirm if it should be &gps_time.tv_sec since gps_time is passed by reference
+
+    //Calculating the UT1 Julian Day from the UTC time and date
+    uint year = utc_time->tm_year + 1900;
+    uint month = utc_time->tm_mon + 1;
+    if (month < 3) {
+        month = month + 12;
+        year = year - 1;
+    }
+    uint day = utc_time->tm_mday;
+    uint hour = utc_time->tm_hour;
+    float JD = 2 - int(year / 100.) + int(int(year / 100.) / 4.) + int(365.25 * year) //See this conversion for Skyfield
+               + int(30.6001 * (month + 1)) + day + 1720994.5; 
+    double UT = (hour) + (utc_time->tm_min / 60.)
+               + (utc_time->tm_sec + _beam_coord.UT1_UTC + gps_time.tv_nsec / 1.e9) / 3600.;                //TO DO:UT1_UTC should be a double value in struct _beam_coord containing UT1-UTC in seconds
```

Comment:
Once the UTC-UT1 call back function is added to the base class, then the value should be used here.

Reply:
Done!